### ISSUE TITLE
Incorrect error 42804 when using a DOMAIN over a composite type in an INOUT function

### ISSUE THREAD

**Issue by mavlsion** [okbob/plpgsql_check#202]
Hello,

I've encountered an issue with the `plpgsql_check` extension when used together with a **DOMAIN defined over a composite type**. I would appreciate clarification on whether this behavior is expected or if it should be fixed.

#### Reproduction Steps

```sql
CREATE TYPE my_composite AS (id numeric(12,0), tag text);

CREATE DOMAIN bad_domain AS my_composite;

CREATE TYPE dept_info AS (dept_id int);

CREATE OR REPLACE FUNCTION test_func(
    INOUT rCheckBuffer bad_domain,
    IN    rDeptInfo    dept_info
)
AS $$
BEGIN
    rCheckBuffer := ROW(123, 'test')::bad_domain;
    RETURN;
END;
$$ LANGUAGE plpgsql;

DO $$
DECLARE
    input_buffer bad_domain;
    input_dept   dept_info;
    result       bad_domain;
BEGIN
    input_buffer := ROW(999, 'initial')::bad_domain;
    input_dept   := ROW(42)::dept_info;

    SELECT * INTO result
    FROM test_func(input_buffer, input_dept);

    RAISE NOTICE 'Result: %', result;
END $$;
```

**Output:**
```sql
NOTICE:  Result: (123,test)
DO
```

However, when checking the function with `plpgsql_check`:

```sql
SELECT * FROM plpgsql_check_function('test_func(bad_domain, dept_info)');
```

**It reports an error:**
```
error:42804: returned record type does not match expected record type
Detail: Returned type numeric(12,0) does not match expected type bad_domain in column "__result__" (position 1).
```

The function works correctly at runtime and assigns a complete composite value compatible with the domain. Yet `plpgsql_check` incorrectly interprets the return type as a scalar (`numeric`) instead of the full composite structure.

**Comment by okbob** [comment]
I wrote caeea7418197887e77cb779f1fa3375a42928acc

This is maybe deeper problem. I found some issues in Postgres.

**Comment by okbob** [comment]
Now, for you examples, you should to get correct warning:

```
SELECT * FROM plpgsql_check_function('test_func(bad_domain, dept_info)');
┌──────────────────────────────────────────────────┐
│              plpgsql_check_function              │
╞══════════════════════════════════════════════════╡
│ warning extra:00000:unused parameter "rdeptinfo" │
└──────────────────────────────────────────────────┘
(1 row)
```
